kinto/core/authorization.py

```diff
@@ -179,9 +183,17 @@ def fetch_shared_records(self, perm, principals, get_bound_permissions):
             return value. The attribute is then read by
             :class:`kinto.core.resource.ShareableResource`
         """
-        ids = self.get_shared_ids(permission=perm,
-                                  principals=principals,
-                                  get_bound_permissions=get_bound_permissions)
+        bound_permissions = []
+        if get_bound_permissions is not None:
+            bound_permissions = [(o, p) for (o, p) in
+                                 get_bound_permissions(self._object_id_match,
+                                                       perm)
+                                 if o.endswith(self._object_id_match)]
```

Comment:
Although this was move from every permission backend implementation to here, I believe it is not properly tested / specified